saga/sagaRecovery_test.go

```diff
@@ -92,7 +92,7 @@ func TestRecoverState_UpdateSagaStateFails(t *testing.T) {
 	sagaLogMock.EXPECT().GetMessages(sagaId).Return(msgs, nil)
 	sc := MakeSagaCoordinator(sagaLogMock)
 
-	state, err := recoverState(sagaId, sc, RollbackRecovery)
```

Comment:
No they were not testing RollbackRecovery, the logic for recovering state and applying the safe state logic used to be in one function, when it got refactored I forgot to remove the RecoveryState

Its now split sagaRecovery recoverState is all the logic to take SagaLog messages and reapply them to create a *SagaState.  SagaCoordinator.RecoverSagaState now handles applying the Recovery Method.